### PR TITLE
efivar bug

### PR DESCRIPTION
**Pull request by frelon**
- **Do not return error for efi.ReadLoadOption**
- **Refactor and test efi manager**

### CODE REVIEW

File: pkg/bootloader/grub.go [hunk 1]
```diff
@@ -339,7 +339,7 @@ func (g *Grub) clearEntry() error {
 // createBootEntry will create an entry in the efi vars for our shim and set it to boot first in the bootorder
 func (g *Grub) CreateEntry(shimName string, relativeTo string, efiVariables eleefi.Variables) error {
 	g.logger.Debugf("Creating boot entry for elemental pointing to shim %s/%s", constants.EntryEFIPath, shimName)
-	bm, err := eleefi.NewBootManagerForVariables(efiVariables)
+	bm, err := eleefi.NewBootManagerForVariables(g.logger, efiVariables)
```
Comment (davidcassany) [inline pkg/bootloader/grub.go:342]: Nice, always good having more accurate logs

File: pkg/efi/manager.go [hunk 2]
```diff
@@ -57,9 +59,10 @@ func NewBootManagerForVariables(efivars Variables) (BootManager, error) {
 		if err != nil {
 			return BootManager{}, fmt.Errorf("cannot read %s: %v", name, err)
 		}
-		entry.LoadOption, err = efi.ReadLoadOption(bytes.NewReader(entry.Data))
+		entry.LoadOption, err = bm.efivars.ReadLoadOption(bytes.NewReader(entry.Data))
 		if err != nil {
-			return bm, err
+			logger.Warnf("Error reading efi load option for '%s': %s", name, err.Error())
+			continue
```
Comment (davidcassany) [inline pkg/efi/manager.go:65]: I guess this is the actual fix. Where do we see this issue? Which is the impact of this bug. I am wondering if this is something to be backported somewhere.

**Review by davidcassany — APPROVED** [review]
Looks good to me,
Just wondering about which use case is this actually fixing.

File: pkg/efi/manager.go (hunk 2 above)
Comment (frelon) [reply pkg/efi/manager.go:65]: Correct, this is from the linked issue, which is coming from [this harvester issue](https://github.com/harvester/harvester/issues/4528)

**Pull request merged**